test/attr/ignore_deprecation_warnings.swift

```diff
@@ -0,0 +1,65 @@
+// RUN: %target-typecheck-verify-swift
+
+@available(*, deprecated)
+func deprecatedFunc() {}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue() -> Int {
+    return 3
+}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue2() -> String {
+    return ""
+}
+
+@available(*, deprecated)
+var deprecatedVariable = ""
+
+@available(*, deprecated)
+struct deprecatedType {}
+
+@ignoreDeprecationWarnings
+func x() {
+    deprecatedFunc()
+}
+
+let instanceThatSouldProduceWarning = deprecatedType() // expected-warning {{'deprecatedType' is deprecated}}
+
+var s = deprecatedVariable // expected-warning {{'deprecatedVariable' is deprecated}}
```

Comment:
Could you add back the tests for top-level and local variables with `@ignoreDeprecationWarnings`, along with FIXME comments to support them?